### PR TITLE
minor fixes

### PR DESCRIPTION
**Pull request by csukuangfj**
## Proposed changes

1. Use `std::lock_guard` to replace `std::unique_lock` when ownership of the mutex does not need to be transferred.
2. Remove trailing `;`

## Checklist

Put an `x` in the boxes that apply.

- [x] I have read the [CONTRIBUTING](https://github.com/ml-explore/mlx/blob/main/CONTRIBUTING.md) document
- [x] I have run `pre-commit run --all-files` to format my code / installed pre-commit prior to committing changes

### CODE REVIEW

File: mlx/event.h [hunk 1]
```diff
@@ -10,46 +10,46 @@ namespace mlx::core {
...
     return event_;
-  };
+  }
 
  private:
   // Default constructed stream should never be used
   // since the event is not yet valid
   Stream stream_{0, Device::cpu};
-  std::shared_ptr<void> event_{nullptr};
+  std::shared_ptr<void> event_;
```
Comment (awni) [inline mlx/event.h:52]: Why remove that?

Comment (csukuangfj) [reply mlx/event.h:52]: shared_ptr is default initialized to nullptr, right?

Comment (awni) [reply mlx/event.h:52]: It looks like it, yes!

**Comment by awni** [comment]
@csukuangfj can you check the tests, looks like the code did not compile.

**Comment by csukuangfj** [comment]
> @csukuangfj can you check the tests, looks like the code did not compile.

Sorry, fixed.

**Review by awni — APPROVED** [review]
LGTM, thanks!

**Pull request merged**